### PR TITLE
[2780] Redirect subject filter page to rails

### PR DESCRIPTION
**Pull request by timabell**
### Context

replacement for accidentally closed #496

Rails is nearly ready to take over this page so prepping the redirect now.

### Changes proposed in this pull request

* Regression test existing behaviour (good practice for working with legacy code)
* TDD redirection to new page

### Guidance to review

```bash
FEATURE_REDIRECT_TO_RAILS=true dotnet run
```

### CODE REVIEW

**Comment by timabell** [comment]
[[S] subjects filter: Redirect from C# to Rails](https://trello.com/c/FQom150q/2780-s-subjects-filter-redirect-from-c-to-rails)

**Comment by timabell** [comment]
rebased

File: tests/Unit.Tests/Controllers/FilterControllerTests.cs [hunk 1]
```diff
@@ -97,6 +101,36 @@ public void StudyTypeRedirectsToNewApp()
...
+        {
+            var mockSubjectAreas = new List<SubjectArea>();
+            _mockApi.Setup(api => api.GetSubjectAreas()).Returns(mockSubjectAreas);
+            var result = _filterController.SubjectGet(_resultsFilter);
+            result.Should().BeOfType<ViewResult>();
+            var viewResult = result as ViewResult;
+            viewResult?.Model.Should().BeOfType<SubjectFilterViewModel>();
+            var model = (SubjectFilterViewModel)viewResult?.Model;
+            model.FilterModel.Should().Be(_resultsFilter);
+            model.SubjectAreas.Should().BeSameAs(mockSubjectAreas);
```
Comment (faissalMT) [inline tests/Unit.Tests/Controllers/FilterControllerTests.cs]: Might be good to have some paragraphing here

**Comment by defong** [comment]
superseded by #509

**Pull request closed without merging**